tests/optimisation_test/recipe_converter.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd
+import yaml
+from archemist.core.persistence.yaml_handler import YamlHandler
+from nested_lookup import nested_update
+from pathlib import Path
+
+class RecipeUpdate:
+    def __init__(self) -> None:
+        self.placeholder_keys = []
+        self.placeholders =[]
+        super().__init__()
+
+    def find_placeholders(self, data):
+        self._data = data
+        if isinstance(self._data, dict):
+            for key, value in self._data.items():
+                if isinstance(value, str) and '{{' in value and '}}' in value:
+                    self.placeholder_keys.append(key)
+                    self.placeholders.append(value)
+                else:
+                    self.find_placeholders(value)
+        elif isinstance(self._data, list):
+            for item in self._data:
+                self.find_placeholders(item)
+             
+    def update_values(self,data, *args):
+        self.data = data
+        _output_pd = args[0]
+        _shape = _output_pd.shape
+        _total_columns = _shape[1]
+        for _key in self.placeholder_keys:
+            for col in range(_total_columns):
+                _head = _output_pd.columns[col]
+                if _key == _head:
+                    _values = _output_pd[_head].tolist()
+                    print(_key,_values)
+                    _data_update = nested_update(self.data, key = str(_key), value = _values)
+                    self.data = _data_update
+                elif _key != _head:
+                    pass      
+
+    def create_optimized_recipe(self, path):
+        with open(path, 'w') as file:
+            print(self.data)
+            yaml.dump(self.data, file)
+
+if __name__ == "__main__": 
+    path = Path.cwd() / "tests" / "optimisation_test" / "algae_bot_recipe_template.yaml"
```

Comment:
For better readability and to keep consistency with past development:
```python
cwd_path = Path.cwd()
recipe_path = Path.joinpath(cwd_path, "test/optimisation_test/algae_bot_recipe_template.yaml"
```

Reply:
changed